### ISSUE TITLE
add support for 18 snapshots

### ISSUE THREAD

**Issue by PortableGames** [CodedSakura/CodedSMPUtils#1]
support for 18 snapshots

**Comment by PortableGames** [comment]
it is possible that fabric's dependecy overrides would solve the malfunction?
https://fabricmc.net/wiki/tutorial:dependency_overrides

**Comment by CodedSakura** [comment]
no, as 1.18 is now using java 16 features (in 1.17 it was still using only java 8 features on java 16). that means that i have to update both Loom and Fabric loader, and quite likely fix all consequent errors (that arise due to new mappings, etc.). i currently don't have enough free time nor will to go through with constantly uptading to 1.18 snapshots... i hope i can find enough time when 1.18 prereleases start appearing, as they don't introduce many changes in mojang's code.

**Comment by PortableGames** [comment]
and didn't fapi 0.40.4 try to solve this problem with the mappings?

**Comment by CodedSakura** [comment]
no, fapi 0.40.4, like almost all fapi updates probably only added the new 21w38a features to fapi. it's not goona be a magic tool that'll fix any and all issues..

**Comment by PortableGames** [comment]
but for example mods like minimotd or itemflexer, fabricvotifier, bluemap and player-events .. work without errors even on the new version .. how is this possible?

**Comment by CodedSakura** [comment]
well, none of the mods you listed depend on a specifc MC version, only fabric API, meanwhile CSMPU is pretty intertwined with 1.17. of course you can always try, it's just that i don't think that it'll be that easy..

**Comment by PortableGames** [comment]
I tried it and it doesn't work .. the server won't turn on

wouldn't it be a good idea to make your mod similar? without depending on the version?

**Comment by PortableGames** [comment]
if you can create a fabric_loader_dependencies.json that will run your mod please send it to me

**Comment by CodedSakura** [comment]
i don't believe a simple fabric_loader_dependencies.json will allow my mod to successfully run on the 1.18 snapshots
